Building Starrocks main is failing on Debian 12

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. I created new Debian 12 VM.
```
maverick@lima-debian-12:~/starrocks$ uname -a
Linux lima-debian-12 6.1.0-28-cloud-arm64 #1 SMP Debian 6.1.119-1 (2024-11-22) aarch64 GNU/Linux
```
2. Then I installed all the packages using this instructions: 
https://docs.starrocks.io/docs/developers/build-starrocks/build_starrocks_on_ubuntu/
3. I have proper versions of the compilers
```
maverick@lima-debian-12:~/starrocks$ cmake -version
cmake version 3.25.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
maverick@lima-debian-12:~/starrocks$ java --version
openjdk 17.0.13 2024-10-15
OpenJDK Runtime Environment (build 17.0.13+11-Debian-2deb12u1)
OpenJDK 64-Bit Server VM (build 17.0.13+11-Debian-2deb12u1, mixed mode, sharing)
maverick@lima-debian-12:~/starrocks$ gcc --version
gcc (Debian 12.2.0-14) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

maverick@lima-debian-12:~/starrocks$ g++ --version
g++ (Debian 12.2.0-14) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
4. I'm using ./build.sh script.

### Expected behavior (Required)
Starrocks will build.

### Real behavior (Required)

Build is failing with following error.
```
-- Build files have been written to: /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/starrocks_build
[  3%] Building CXX object CMakeFiles/flatbuffers.dir/src/code_generators.cpp.o
[  6%] Building CXX object CMakeFiles/flatc.dir/src/code_generators.cpp.o
[ 12%] Building CXX object CMakeFiles/flatbuffers.dir/src/idl_parser.cpp.o
[ 12%] Building CXX object CMakeFiles/flatc.dir/src/idl_parser.cpp.o
In file included from /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/idl.h:25,
                 from /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/src/idl_parser.cpp:23:
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Object::KeyCompareLessThan(const reflection::Object*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:659:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Object]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Object>*; _Iterator2 = flatbuffers::Offset<reflection::Object>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/idl.h:25,
                 from /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/src/idl_parser.cpp:23:
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Object::KeyCompareLessThan(const reflection::Object*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:659:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Object]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Object>*; _Iterator2 = flatbuffers::Offset<reflection::Object>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Object::KeyCompareLessThan(const reflection::Object*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:659:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Object]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Object>*; _Iterator2 = flatbuffers::Offset<reflection::Object>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Object::KeyCompareLessThan(const reflection::Object*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:659:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Object]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Object>*; _Iterator2 = flatbuffers::Offset<reflection::Object>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Object>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Enum::KeyCompareLessThan(const reflection::Enum*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:351:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Enum]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Enum>*; _Iterator2 = flatbuffers::Offset<reflection::Enum>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Enum::KeyCompareLessThan(const reflection::Enum*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:351:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Enum]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Enum>*; _Iterator2 = flatbuffers::Offset<reflection::Enum>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Enum::KeyCompareLessThan(const reflection::Enum*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:351:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Enum]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Enum>*; _Iterator2 = flatbuffers::Offset<reflection::Enum>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Enum::KeyCompareLessThan(const reflection::Enum*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:351:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Enum]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Enum>*; _Iterator2 = flatbuffers::Offset<reflection::Enum>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Enum>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Service::KeyCompareLessThan(const reflection::Service*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:903:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Service]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Service>*; _Iterator2 = flatbuffers::Offset<reflection::Service>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Service::KeyCompareLessThan(const reflection::Service*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:903:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Service]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Service>*; _Iterator2 = flatbuffers::Offset<reflection::Service>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Service::KeyCompareLessThan(const reflection::Service*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:903:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Service]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Service>*; _Iterator2 = flatbuffers::Offset<reflection::Service>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Service::KeyCompareLessThan(const reflection::Service*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:903:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Service]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Service>*; _Iterator2 = flatbuffers::Offset<reflection::Service>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Service>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Field::KeyCompareLessThan(const reflection::Field*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:481:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Field]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Field>*; _Iterator2 = flatbuffers::Offset<reflection::Field>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Field::KeyCompareLessThan(const reflection::Field*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:481:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Field]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Field>*; _Iterator2 = flatbuffers::Offset<reflection::Field>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Field::KeyCompareLessThan(const reflection::Field*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:481:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Field]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Field>*; _Iterator2 = flatbuffers::Offset<reflection::Field>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::Field::KeyCompareLessThan(const reflection::Field*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:481:20,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::Field]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::Field>*; _Iterator2 = flatbuffers::Offset<reflection::Field>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::Field>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::KeyValue::KeyCompareLessThan(const reflection::KeyValue*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:170:19,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::KeyValue]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::KeyValue>*; _Iterator2 = flatbuffers::Offset<reflection::KeyValue>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::KeyValue::KeyCompareLessThan(const reflection::KeyValue*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:170:19,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::KeyValue]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::KeyValue>*; _Iterator2 = flatbuffers::Offset<reflection::KeyValue>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1867:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::KeyValue::KeyCompareLessThan(const reflection::KeyValue*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:170:19,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::KeyValue]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::KeyValue>*; _Iterator2 = flatbuffers::Offset<reflection::KeyValue>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
In member function ‘bool flatbuffers::String::operator<(const flatbuffers::String&) const’,
    inlined from ‘bool reflection::KeyValue::KeyCompareLessThan(const reflection::KeyValue*) const’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/reflection_generated.h:170:19,
    inlined from ‘bool flatbuffers::FlatBufferBuilder::TableKeyComparator<T>::operator()(const flatbuffers::Offset<T>&, const flatbuffers::Offset<T>&) const [with T = reflection::KeyValue]’ at /home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:1608:41,
    inlined from ‘bool __gnu_cxx::__ops::_Iter_comp_iter<_Compare>::operator()(_Iterator1, _Iterator2) [with _Iterator1 = flatbuffers::Offset<reflection::KeyValue>*; _Iterator2 = flatbuffers::Offset<reflection::KeyValue>*; _Compare = flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue>]’ at /usr/include/c++/12/bits/predefined_ops.h:158:30,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1870:17,
    inlined from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1889:40,
    inlined from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = flatbuffers::Offset<reflection::KeyValue>*; _Size = long int; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue> >]’ at /usr/include/c++/12/bits/stl_algo.h:1921:38:
/home/maverick.linux/starrocks/thirdparty/src/flatbuffers-1.10.0/include/flatbuffers/flatbuffers.h:356:18: error: ‘int strcmp(const char*, const char*)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  356 |     return strcmp(c_str(), o.c_str()) < 0;
      |            ~~~~~~^~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/flatbuffers.dir/build.make:90: CMakeFiles/flatbuffers.dir/src/idl_parser.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/flatbuffers.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/flatc.dir/build.make:90: CMakeFiles/flatc.dir/src/idl_parser.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 15%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_text.cpp.o
make[1]: *** [CMakeFiles/Makefile2:113: CMakeFiles/flatc.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### StarRocks version (Required)
```
maverick@lima-debian-12:~/starrocks$ git log -n 1
commit f977337f8492eee4ab540aa9713f96802e91a971 (HEAD -> main, origin/main, origin/HEAD)
Author: xiangguangyxg <110401425+xiangguangyxg@users.noreply.github.com>
Date:   Fri Dec 27 15:49:37 2024 +0800

    [Feature] Support restoring from a cluster snapshot for shared-data mode (part 3, introduce gtid for tablet metadata) (#54326)

    Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>
 ```

It may have no sense as I'm getting more errors.

```
make[1]: Leaving directory '/home/maverick.linux/starrocks/thirdparty/src/libfiu-1.1/utils'
===== begin build llvm-project-16.0.6.src
-- Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
-- Could NOT find LibXml2 (missing: LIBXML2_LIBRARY LIBXML2_INCLUDE_DIR)
-- Could NOT find LibEdit (missing: LibEdit_INCLUDE_DIRS LibEdit_LIBRARIES)
-- Native target architecture is AArch64
-- Threads enabled.
-- Doxygen disabled.
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH)
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH)
-- OCaml bindings disabled.
-- LLVM host triple: aarch64-unknown-linux-gnu
-- LLVM default target triple: aarch64-unknown-linux-gnu
-- Building with -fPIC
-- Targeting AArch64
-- Configuring done
-- Generating done
-- Build files have been written to: /home/maverick.linux/starrocks/thirdparty/src/llvm-project-16.0.6.src
make: *** No rule to make target 'LLVMBitstreamReader'.  Stop.
```

I will try to switch to official docker method.

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -80,18 +80,6 @@ check_prerequest() {
     fi
 }
 
-# echo if gcc version is greater than 14.0.0
-# else echo ""
-echo_gt_gcc14() {
-    local version=$($CC --version | grep -oP '(?<=\s)\d+\.\d+\.\d+' | head -1)
-    if [[ $(echo -e "14.0.0\n$version" | sort -V | tail -1) == "14.0.0" ]]; then
-        echo ""
-    else
-        #gt gcc14
-        echo "$1"
-    fi
-}
-
 # sudo apt-get install cmake
 # sudo yum install cmake
 check_prerequest "${CMAKE_CMD} --version" "cmake"
@@ -599,7 +587,7 @@ build_rocksdb() {
 
     CFLAGS= \
     EXTRA_CFLAGS="-I ${TP_INCLUDE_DIR} -I ${TP_INCLUDE_DIR}/snappy -I ${TP_INCLUDE_DIR}/lz4 -L${TP_LIB_DIR} ${FILE_PREFIX_MAP_OPTION}" \
-    EXTRA_CXXFLAGS=$(echo_gt_gcc14 -Wno-error=redundant-move)" -fPIC -Wno-deprecated-copy -Wno-stringop-truncation -Wno-pessimizing-move -I ${TP_INCLUDE_DIR} -I ${TP_INCLUDE_DIR}/snappy ${FILE_PREFIX_MAP_OPTION}" \
+    EXTRA_CXXFLAGS="-fPIC -Wno-redundant-move -Wno-deprecated-copy -Wno-stringop-truncation -Wno-pessimizing-move -I ${TP_INCLUDE_DIR} -I ${TP_INCLUDE_DIR}/snappy ${FILE_PREFIX_MAP_OPTION}" \
     EXTRA_LDFLAGS="-static-libstdc++ -static-libgcc" \
     PORTABLE=1 make USE_RTTI=1 -j$PARALLEL static_lib
 
@@ -664,8 +652,9 @@ build_flatbuffers() {
   cd $BUILD_DIR
   rm -rf CMakeCache.txt CMakeFiles/
 
-  export CXXFLAGS="-O3 -fno-omit-frame-pointer -fPIC -g "$(echo_gt_gcc14 "-Wno-error=stringop-overread")
-  export CPPFLAGS="-O3 -fno-omit-frame-pointer -fPIC -g "$(echo_gt_gcc14 "-Wno-error=stringop-overread")
+  # reset the optimization level to O2 to avoid "stringop-overread" warnings/errors
+  export CXXFLAGS="$CXXFLAGS -O2"
+  export CPPFLAGS="$CPPFLAGS -O2"
 
   LDFLAGS="-static-libstdc++ -static-libgcc" \
   ${CMAKE_CMD} .. -G "${CMAKE_GENERATOR}" -DFLATBUFFERS_BUILD_TESTS=OFF

